### PR TITLE
[CI:DOCS] Build and Sign Mac Pkginstaller

### PR DESCRIPTION
**Pull request by ashley-cui**
Create a new GitHub Action that builds and signs the Mac pkginstaller. The action also uploads the installers to the release, and updates the shasums file.

This action is triggered on release creation, but it can also be triggered manually via a workflow dispatch.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by ashley-cui** [comment]
@cevich PTAL 😌

**Comment by rhatdan** [comment]
LGTM

File: .github/workflows/mac-pkg.yml
Comment (cevich) [inline .github/workflows/mac-pkg.yml:61]: I think maybe this conditional isn't needed.  Having it could be confusing to future maintainers.  If I'm reading it wrong and it _is_ needed, please consider adding a comment.

File: .github/workflows/mac-pkg.yml
Comment (cevich) [inline .github/workflows/mac-pkg.yml:66]: Same here, I think maybe go should always be setup?

File: .github/workflows/mac-pkg.yml
Comment (cevich) [inline .github/workflows/mac-pkg.yml:69]: Defining this hear seems it may be a maintenance headache.  In other words, maintainers are highly likely to forget to update this version when necessary/appropriate.  It will also be difficult to coordinate with the version used in CI VM's by cirrus - meaning what we test may not == what we build :disappointed:   Lastly, it may cause problems if a maintainer does a manual run of the job, specifying a `version` input that's incompatible with the hard-coded golang version :cry: 

Off-hand I don't have a good solution for this that's not complex or likely to flake due to job synchronization with Cirrus.  Maybe it's not a big deal with MacOS builds, but I can say from experience the damn golang version changes about every time with new Fedora CI VM images.  Maybe it's something that can be fixed in a follow-up PR?  Maybe a golang expert knows a clever way we could solve this?

For reference, the "simpelest" (imperfect) solution I can think of is to add steps that:
1. Run `podman version` in `quay.io/podman/upstream:latest`
2. Extract the golang version from the output
3. Use that value as the input to the `actions/setup-go` action

Note: This is imperfect because 'latest' is unlikely to coorespond to the MacOS version being built in this job.  However, maybe the gloang version used is "close enough"?

---

Slightly more complex / more accurate (still imperfect) solution:

1. Look up the `IMAGE_SUFFIX` from the checked out `.cirrus.yml` file
2. run `go env GOVERSION` in `quay.io/libpod/fedora_podman:$IMAGE_SUFFIX`
3. parse and extract the golang version from the output.
4. Use that value as the input to the `actions/setup-go` action

File: .github/workflows/mac-pkg.yml
Comment (cevich) [inline .github/workflows/mac-pkg.yml:83]: Github will "try" to mask secret values.  Including in any output/error messages from the tools.  It's not 100% perfect.  e.g. if an error causes a secret to line-wrap, it may not be hidden.  I suggest spending some time going over this with a fine-tooth-comb to make sure no leaks to the console are possible.  Maybe even adding some (very debug frustrating) `&> /dev/null` (assuming mac-bash supports that) where appropriate.

**Review by cevich — CHANGES_REQUESTED** [review]
Innitial pass, really nice job Ashley.  This is super tricky stuff (much more than it looks by just the YAML).  Made a few comments, and would like to (perhapse) have a team discussion about the golang-version question.

File: .github/workflows/mac-pkg.yml
Comment (ashley-cui) [reply .github/workflows/mac-pkg.yml:66]: Ditto

File: .github/workflows/mac-pkg.yml
Comment (ashley-cui) [reply .github/workflows/mac-pkg.yml:69]: Would it be enough to use the go version specified in the go.mod? That would be pretty easy to extract and set as the go version.

File: .github/workflows/mac-pkg.yml
Comment (ashley-cui) [reply .github/workflows/mac-pkg.yml:83]: Good idea, I'll go through this again and trigger some failures to see if anything pops up.

File: .github/workflows/mac-pkg.yml
Comment (cevich) [reply .github/workflows/mac-pkg.yml:69]: Note: I see we have this same problem in `.github/workflows/upload-win-installer.yml` as well.

> the go version specified in the go.mod

I've considered this in the past.  What I found is, it's a specification not a discrete value.  It also reflects the _minimum_ version, not the one we tested with.

I also considered teaching renovate to manage the version.  That would keep it updated, but not necessarily in-sync with what we test.

This might just be one of those super-thorny problems it's better to stick your head in the sand for now.  Clearly it will need fixing in two workflows, so maybe that can be done in a future PR.

File: .github/workflows/mac-pkg.yml
Comment (cevich) [reply .github/workflows/mac-pkg.yml:83]: Thanks.  I'm not worried so much about somebody hacking the script in a PR (it won't work).  I think the main threat is from accidental output due to some unforeseen runtime condition.

File: .github/workflows/mac-pkg.yml
Comment (cevich) [inline .github/workflows/mac-pkg.yml:2]: Minor: At one point, there was a big TODO made by github re: adding [a top-level `permissions` block](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions).  But it's ugly, complex to figure out, and maybe the defaults are okay?  The Windows workflow doesn't have one either, so :man_shrugging:

File: .github/workflows/mac-pkg.yml
Comment (cevich) [reply .github/workflows/mac-pkg.yml:61]: Thanks, that will be a helpful reminder.  I see that all subsequent steps also have conditions, so this should be fine.

File: .github/workflows/mac-pkg.yml
Comment (cevich) [reply .github/workflows/mac-pkg.yml:69]: Thinking more about this.  Let's worry about it later.  I can start a discussion on the topic to see if perhaps somebody has a simple way to solve it - or can confirm it's a non-problem.

Comment (cevich) [reply .github/workflows/mac-pkg.yml:69]: Opened https://github.com/containers/podman/discussions/19404

File: .github/workflows/mac-pkg.yml
Comment (ashley-cui) [reply .github/workflows/mac-pkg.yml]: Updated and  /dev/null'd the critical commands

File: .github/workflows/mac-pkg.yml
Comment (ashley-cui) [reply .github/workflows/mac-pkg.yml:2]: Added, thanks!

**Comment by ashley-cui** [comment]
@cevich Updated, PTanotherL, thanks for the thorough reviews!

**Review by vrothberg — APPROVED** [review]
LGTM assuming @cevich's content :)

**Review by cevich — APPROVED** [review]
/lgtm

**Comment by cevich** [comment]
@ashley-cui is there a way you can do a manual test-run of this after it merges?  It would be good to identify any problems before it's time to do a real release.

**Comment by ashley-cui** [comment]
@cevich I can, it requires me to mess with some of the existing artifacts on a github release page though, I think the best one to mess with is probably 4.6.0-rc1?

**Comment by cevich** [comment]
Maybe a better choice would be to make a really obvious "testing_do_not_use" release?  Up to you, point is to make sure there are no surprises since this job runs so infrequently.

**Comment by cevich** [comment]
In fact...that gives me an idea.  Maybe we should have a "dry-run" mode for these workflows, where they do everything except actually uploading the files.  I can work on that.

**Comment by ashley-cui** [comment]
I think it requires a tag so a testing release won't work correctly. I'm not opposed to just waiting for the next release to go out and seeing if it works or not, because worst case we just end up building the binaries manually. A dry-run mode sounds good to me, but it also requires removing the files already exists check.

**Comment by cevich** [comment]
Though normally not advised, it's possible to delete tags and releases.  So you could create a tag called, `v6.6.6-testing` (just a silly example) "release" to it, and delete it all after.  

I could be okay with waiting.  It makes me slightly nervous, because a knowledgable/caring person doing the release may not have github-action workflow help available to fix it.  But maybe it's okay that the MacOS/Windows binaries are uploaded later?

**Comment by ashley-cui** [comment]
I'll just run this once on 4.6.0-RC1, if something goes wrong I'll just re-upload the existing binaries.

**Comment by ashley-cui** [comment]
Update: Ran on RC1, looks like it did everything correctly!

**Comment by cevich** [comment]
Great! Thanks for checking.  Opened https://github.com/containers/podman/pull/19418 (but I need to test it somehow).

**Pull request merged**
